### PR TITLE
[Chore] Add MUI

### PR DESCRIPTION
**Pull request by gertie-sheshe**
## Description

Add MaterialUI integration.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/ui/package.json [hunk 1]
```diff
@@ -5,5 +5,10 @@
   "license": "MIT",
   "devDependencies": {
     "config": "workspace:*"
+  },
+  "dependencies": {
+    "@emotion/react": "^11.8.2",
+    "@emotion/styled": "^11.8.1",
+    "@mui/material": "^5.5.0"
```
Comment (kilemensi) [inline packages/ui/package.json:12]: From the [docs](https://mui.com/getting-started/installation/#npm):

> Please note that [react](https://www.npmjs.com/package/react) >= 17.0.0 and [react-dom](https://www.npmjs.com/package/react-dom) >= 17.0.0 are peer dependencies.

I don't see where we're adding `react`/`react-dom` as peer or full dependencies. Doesn't that cause any issues/warnings when building?

**Review by kilemensi — COMMENTED** [review]
👍🏽

**Review by nerdlet — APPROVED** [review]
Looks good 🚀 @gertie-sheshe

File: packages/ui/package.json [hunk 2]
```diff
@@ -5,5 +5,13 @@
   "license": "MIT",
   "devDependencies": {
     "config": "workspace:*"
+  },
+  "dependencies": {
+    "@babel/core": "^7.0.0",
```
Comment (kilemensi) [inline packages/ui/package.json:10]:
```suggestion
  "peerDependencies": {
```

This is to make sure a single version of all these dependencies will be used in each app.

**Review by kilemensi — COMMENTED** [review]
👍🏽 

----

1. Update the Button in the UI to include mui Button. How else would we know it works? For example:
```jsx
import * as React from "react";
import MuiButton from "@mui/material/Button";

export const Button = ({ children, ...props }) => {
  return (
    <MuiButton variant="contained" {...props}>
      {children || "Boop"}
    </MuiButton>
  );
};
```
3. Make sure each app (in `apps/` folder) defines it's own dependencies to include mui, etc. For example:
```json
  "dependencies": {
    "@babel/core": "^7.0.0",
    "@emotion/react": "^11.8.2",
    "@emotion/styled": "^11.8.1",
    "@mui/material": "^5.5.0",
    "next": "^12.1.0",
    "react": "^17.0.0",
    "react-dom": "^17.0.0",
    "ui": "workspace:*"
  }
```

File: apps/web/package.json [hunk 3]
```diff
@@ -10,9 +10,10 @@
   },
   "dependencies": {
     "next": "^12.1.0",
-    "react": "^17.0.2",
+    "react": "17.0.0",
     "react-dom": "^17.0.2",
-    "ui": "workspace:*"
+    "ui": "workspace:*",
+    "web": "workspace:^1.0.0"
```
Comment (kilemensi) [inline apps/web/package.json:16]:
```suggestion
```
Any reason why `web` app would depend on `web` again?

File: packages/config/.babelrc [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
+  "presets": ["next/babel"],
+  "plugins": []
+}
```
Comment (kilemensi) [inline packages/config/.babelrc:4]: Do we need this / is it being used anywhere?

**Review by kilemensi — COMMENTED** [review]
👍🏽 

-----

Docs up is failing:

Screenshot from 2022-03-14 11-33-08

File: apps/web/package.json (hunk 3 above)
Comment (gertie-sheshe) [reply apps/web/package.json:16]: Me I am confusion 😂

**Comment by gertie-sheshe** [comment]
> 👍🏽
> 
> Docs up is failing:
> 
> Screenshot from 2022-03-14 11-33-08

This would happen due to different React versions..hmm

File: packages/ui/package.json [hunk 5]
```diff
@@ -5,5 +5,15 @@
   "license": "MIT",
   "devDependencies": {
     "config": "workspace:*"
+  },
+  "dependencies": {
+    "@babel/core": "^7.17.7",
+    "@emotion/react": "^11.8.2",
+    "@emotion/styled": "^11.8.1",
+    "@mui/material": "^5.5.1"
```
Comment (kilemensi) [inline packages/ui/package.json:13]: Why aren't these `peerDependencies` as well?


----

For context, say you are using this dependency and some other library that also relies on MUI, e.g. formik-mui. This means if both libraries include MUI, we'll now have 2 MUI libraries bundled together.

File: packages/ui/package.json [hunk 6]
```diff
@@ -5,5 +5,15 @@
...
+  },
+  "peerDependencies": {
+    "react": "17.0.0",
+    "react-dom": "17.0.0"
```
Comment (kilemensi) [inline packages/ui/package.json:17]:
```suggestion
    "react": "^17.0.0",
    "react-dom": "^17.0.0"
```

**Review by kilemensi — COMMENTED** [review]
👍🏽

File: apps/web/package.json [hunk 7]
```diff
@@ -10,8 +10,6 @@
   },
   "dependencies": {
     "next": "^12.1.0",
-    "react": "^17.0.2",
-    "react-dom": "^17.0.2",
```
Comment (kilemensi) [inline apps/web/package.json:14]: Leave these here since they're only declared as `peerDependencies` in `ui`.

**Review by kilemensi — APPROVED** [review]
🚀

**Pull request merged**